### PR TITLE
[Merged by Bors] - feat: port Order.Heyting.Boundary

### PR DESCRIPTION
**Pull request by rosborn**
mathlib SHA: 70d50ecfd4900dd6d328da39ab7ebd516abe4025

Porting notes:
1. Needed to fix the proof of `boundary_le_boundary_sup_sup_boundary_inf_left` as simp produced a different term than in mathlib3
2. The `scoped` notation doesn't automatically introduce the notation into the current scope.
~~3. The lint error seems like a bug. Can anyone confirm?~~

### CODE REVIEW

File: Mathlib/Order/Heyting/Boundary.lean [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+variable [CoheytingAlgebra α] {a b : α}
+
+/-- The boundary of an element of a co-Heyting algebra is the intersection of its Heyting negation
+with itself. Note that this is always `⊥` for a boolean algebra. -/
+def boundary (a : α) : α :=
+  a ⊓ ￢a
+#align coheyting.boundary Coheyting.boundary
+
+-- mathport name: «expr∂ »
+scoped[Heyting] prefix:120 "∂ " => Coheyting.boundary
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Heyting/Boundary.lean:39]:
```suggestion
/-- The boundary of an element of a co-Heyting algebra. -/
scoped[Heyting] prefix:120 "∂ " => Coheyting.boundary
```

**Comment by rosborn** [comment]
Should I rename `hnot` to `hNot` in this PR?

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**